### DISCUSSION TITLE
Getting status HostNotFound when using hostname instead of IP

### DISCUSSION THREAD

**Discussion by rofr** [sipsorcery-org/sipsorcery#929] [Q&A]
The following call
```csharp
var status = await SipTransport.SendRequestAsync(sipRequest, waitForDns: true);
```
fails with a HostNotFound status when using a dns hostname in the sip uri. IPAddress works fine. Yes, the hostname resolves on the machine that this code runs on. I was building the uri with custom code and doing a dns lookup to get an ip before I discovered the `SIPURI.ParseSIPURI()` method. And I'm trying hard to understand the purpose of the waitForDns argument. If we need to lookup an ip from the host name, how can we not wait for the result?

Can anyone 
Building the request using the code below and passing endpoints in the format: `sip:user@host:port`

```csharp
 private static SIPRequest BuildSipRequest(bool useFailover, ScaipRequest scaipRequest, Dictionary<string, object> logRecord)
 {
      var uri = useFailover
          ? SIPURI.ParseSIPURI(scaipRequest.FailoverEndPoint)
          : SIPURI.ParseSIPURI(scaipRequest.PrimaryEndpoint);

      var sipRequest = SIPRequest.GetRequest(SIPMethodsEnum.MESSAGE, uri);
      logRecord["cseq"] = _cSequenceNumber;
      sipRequest.Header.CSeq = _cSequenceNumber++;
      sipRequest.Header.UserAgent = "redacted/1.0";
      sipRequest.Body = scaipRequest.Xml;
      sipRequest.Header.ContentType = "application/scaip+xml";
      logRecord["sipRequest"] = sipRequest.ToString();
      return sipRequest;
 }
```

**Comment by Rjevski** [comment]
> And I'm trying hard to understand the purpose of the waitForDns argument. If we need to lookup an ip from the host name, how can we not wait for the result?

Don't take my word for it, as I only started seriously using with the library yesterday, but I think that one is intended to be used internally by SIPSorcery - its SIPTransport class implements retransmissions and relies on that to retry; thus the initial packet can trigger a DNS lookup (which will populate the DNS cache) but does not need to actively wait for it, instead it can fail early with an error, but the next retransmission (now with a populated DNS cache) will succeed.

However it does seem like something else is broken; I am experiencing the same problem, even when setting `DisableRetransmitSending` on the `SIPTransport`, which according to the code should force "wait for DNS" to be true. If it matters, I am using .NET Core on macOS and Linux.

**Comment by Rjevski** [comment]
I am still stuck on this problem so I ended up opening an actual GitHub issue with a minimal example: https://github.com/sipsorcery-org/sipsorcery/issues/980

**Comment by Rjevski** [comment]
Seems to be a bug introduced in 6.0.12? I am not experiencing this problem on 6.0.11 so I suggest downgrading until it's fixed.
